### DISCUSSION TITLE
关于b站“自动开播”功能的建议

### DISCUSSION THREAD

**Discussion by YaeSakuraQvQ** [the1812/Bilibili-Evolved#2085] [功能建议 / Ideas]
您好，希望作者大大您可以增加一个默认关闭b站“自动开播”的功能（自动开播 即打开视频网页后视频自动播放）
1.有时候在打开视频需要先翻评论或收藏或打开多个视频网页屯着 所以不希望其自动播放
2.在使用chrome无痕的时候第一次点开视频会自动播放并伴随b站自动的66%音量（耳机党直接阵亡）
作者大大辛苦了

**Comment by Coulomb-G** [comment]
那你把b站的自动开播关了不就好了?

**Comment by the1812** [comment]
![image](https://user-images.githubusercontent.com/26504152/126903072-b1f75210-e583-43e4-9117-643bcab58d45.png)

**Reply by YaeSakuraQvQ** [reply]
谢谢作者大大回复...我知道这个开关，我的意思是能否让脚本支持把“自动开播”默认置于关闭状态，现在我每天第一次打开b站都需要把“自动开播”关掉,如果可以实现自动化就好了(我在讨论区也看到了类似的...应该不是只有我自己有这种需求吧)
如果从代码角度实现不困难的话还是希望作者大大整一下的，辛苦了

**Reply by Geeyun-JY3** [reply]
你的使用场景是不登录对吧

**Reply by YaeSakuraQvQ** [reply]
使用场景是登录（但是 是chrome无痕）...登录b站账号并没有记住这个“自动开播”的设置

**Reply by Geeyun-JY3** [reply]
> 使用场景是登录（但是 是chrome无痕）...登录b站账号并没有记住这个“自动开播”的设置

那就是开无痕后再登录……那就如你说的第一次打开总会自动播放一次，登录记不记住都改变不了

看看 @the1812 怎么看这个需求吧

**Comment by the1812** [comment]
你自己另外装这个脚本吧
```js
// ==UserScript==
// @name         禁止自动播放
// @version      1.0.0
// @author       Grant Howard
// @license      MIT
// @match        *://www.bilibili.com/*
// @run-at       document-start
// ==/UserScript==

const settings = JSON.parse(localStorage.getItem('bilibili_player_settings'))
settings.video_status = { ...settings.video_status, autoplay: false, autoplay_has_set: true, autoplay_reddot_status: false }
localStorage.setItem('bilibili_player_settings', JSON.stringify(settings))
```

**Reply by YaeSakuraQvQ** [reply]
阔以阔以，谢谢作者大大 万分感谢~
